### PR TITLE
[Feature] Add `databricks_budget` resource

### LINKED ISSUE #3887: [FEATURE] Add support for Budgets
As [Budgets API](https://docs.databricks.com/api/account/budgets/list) entered the public preview phase, it would be good to add corresponding resources.

### PR DESCRIPTION
**Pull request by 840**
## Changes
Add support for account-level [Budget](https://docs.databricks.com/api/account/budgets) resource

Example structure
```
resource "databricks_budget" "this" {
    display_name = "data-science-budget"

    alert_configurations {
        time_period         = "MONTH"
        trigger_type        = "CUMULATIVE_SPENDING_EXCEEDED" 
        quantity_type       = "LIST_PRICE_DOLLARS_USD"
        quantity_threshold  = "840"

        action_configurations {
            action_type = "EMAIL_NOTIFICATION"
            target      = "me@databricks.com"
        }
    }

    filter {
        workspace_id {
            operator = "IN"
            values   = [
                1234567890098765
            ]                              
        }
        tags {
            key   = "Databricks"
            value {
                operator = "IN"
                values = ["Data Science"]
            }
        }
    }
}
```

Resolves #3887

## Tests
- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by alexott** [comment]
Can we ask the product team why `quantity_threshold` is string?

File: mws/resource_mws_budgets.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	Target     string                          `json:"target,omitempty"`
+}
+
+type AlertConfiguration struct {
+	ActionConfigurations []ActionConfiguration                  `json:"action_configurations,omitempty"`
+	QuantityThreshold    string                                 `json:"quantity_threshold,omitempty"`
+	QuantityType         billing.AlertConfigurationQuantityType `json:"quantity_type,omitempty"`
+	TimePeriod           billing.AlertConfigurationTimePeriod   `json:"time_period,omitempty"`
+	TriggerType          billing.AlertConfigurationTriggerType  `json:"trigger_type,omitempty"`
+}
```
Comment (alexott) [inline mws/resource_mws_budgets.go:36]: We need to use structs from SDK

**Review by alexott — CHANGES_REQUESTED** [review]

File: mws/resource_mws_budgets.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			common.DataToStructPointer(d, s, &create)
+			acc, err := c.AccountClientWithAccountIdFromConfig(d)
+			if err != nil {
+				return err
+			}
+			budget, err := acc.Budgets.Create(ctx, billing.CreateBudgetConfigurationRequest{Budget: create})
+			if err != nil {
+				return err
+			}
+			d.SetId(budget.Budget.BudgetConfigurationId)
```
Comment (alexott) [inline mws/resource_mws_budgets.go:32]: We don't need this line - when you call `Pack` it will call `SetId`...

File: mws/resource_mws_budgets.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+func ResourceMwsBudget() common.Resource {
+	s := common.StructToSchema(billing.BudgetConfiguration{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		common.CustomizeSchemaPath(m, "display_name").SetValidateFunc(validation.StringLenBetween(1, 128))
+		for _, p := range []string{"account_id", "budget_configuration_id", "create_time", "update_time"} {
+			common.CustomizeSchemaPath(m, p).SetComputed()
+		}
+		return m
+	})
+	p := common.NewPairSeparatedID("account_id", "budget_configuration_id", "/")
```
Comment (alexott) [inline mws/resource_mws_budgets.go:19]: What is the reason for using `/` as separator? We can just use `NewPairID` instead that will use `|` as separator.

File: mws/resource_mws_budgets.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var create billing.CreateBudgetConfigurationBudget
+			common.DataToStructPointer(d, s, &create)
+			acc, err := c.AccountClientWithAccountIdFromConfig(d)
```
Comment (alexott) [inline mws/resource_mws_budgets.go:24]: why not `c.AccountClient()` ? We should take `account_id` from the provider config, not from resource config.

**Review by alexott — CHANGES_REQUESTED** [review]
Most important comment is about `AccountClientWithAccountIdFromConfig`

File: mws/resource_mws_budget.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+			}
+			return acc.Budgets.DeleteByBudgetId(ctx, id)
+		},
+		Schema: s,
+	}
+}
+
+// API always return a BigDecimal for QuantityThreshold in string representation.
+// To avoid drift, we convert the provided number to BigDecimal in string representation
+// when making requests to the API.
```
Comment (840) [inline mws/resource_mws_budget.go:114]: Have to change this logic to instead use `SetCustomSuppressDiff` for `alert_configurations.#.quantity_threshold`

**Review by alexott — APPROVED** [review]
code lgtm

File: docs/resources/budget.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+---
+subcategory: "Deployment"
+---
+# databricks_mws_budget Resource
+
+-> **Note** Initialize provider with `alias = "account"`, `host = "https://accounts.azuredatabricks.net"` and use `provider = databricks.account` for all `databricks_mws_*` resources.
```
Comment (alexott) [inline docs/resources/budget.md:6]: We don't have budgets on Azure yet.

Comment (840) [reply docs/resources/budget.md]: Changed to the AWS equivalent `https://accounts.cloud.databricks.com`

**Comment by alexott** [comment]
I put docs into `FinOps` category to distinguish it from deployment-related docs

**Comment by alexott** [comment]
Integration tests are failing on GCP, although they should pass...

**Comment by alexott** [comment]
@tanmay-db can we get review for this?

**Comment by tanmay-db** [comment]
Hi @alexott , thanks for the ping, I will take a look

File: internal/acceptance/budget_test.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package acceptance
+
+import "testing"
+
+func TestMwsAccBudget(t *testing.T) {
+	loadAccountEnv(t)
+	AccountLevel(t, Step{
```
Comment (tanmay-db) [inline internal/acceptance/budget_test.go:7]: We should add another step to test the update pathway

File: finops/resource_budget_test.go [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+		Resource: ResourceBudget(),
+	}.ApplyAndExpectData(t, map[string]any{
+		"display_name":           "budget_name",
+		"id":                     "account_id|budget_configuration_id",
+		"alert_configurations.#": 1,
+		"filter.#":               1,
+	})
+}
+
+func TestResourceBudgetRead(t *testing.T) {
```
Comment (tanmay-db) [inline finops/resource_budget_test.go:124]: Might be good to add some more tests to check the error pathways like we have for other resource eg: unpack(d) leading to error (i.e. id not supplied properly).

File: finops/resource_budget.go [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+func ResourceBudget() common.Resource {
+	s := common.StructToSchema(billing.BudgetConfiguration{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		common.CustomizeSchemaPath(m, "display_name").SetValidateFunc(validation.StringLenBetween(1, 128))
+		for _, p := range []string{"account_id", "budget_configuration_id", "create_time", "update_time"} {
+			common.CustomizeSchemaPath(m, p).SetComputed()
+		}
+		common.CustomizeSchemaPath(m, "alert_configurations", "alert_configuration_id").SetComputed()
+		common.CustomizeSchemaPath(m, "alert_configurations", "action_configurations", "action_configuration_id").SetComputed()
+		common.CustomizeSchemaPath(m, "alert_configurations", "quantity_threshold").SetCustomSuppressDiff(func(k, old, new string, d *schema.ResourceData) bool {
```
Comment (tanmay-db) [inline finops/resource_budget.go:21]: Might be good to add a small comment on what this suppress diff is doing.

**Pull request merged**